app/modules/node.c

```diff
@@ -76,34 +77,27 @@ static int node_deepsleep( lua_State* L )
 {
   uint64 us;
   uint8 option;
-  //us = luaL_checkinteger( L, 1 );
   // Set deleep option, skip if nil
   if ( lua_isnumber(L, 2) )
   {
-    option = lua_tointeger(L, 2);
-    if ( option < 0 || option > 4)
-      return luaL_error( L, "wrong arg range" );
-    else
-      system_deep_sleep_set_option( option );
+    luaL_argcheck(L, 2, (option = lua_tounsigned(L, 2)) <= 4, "wrong option value" );
```

Comment:
Yick!  I understand that that's going to compile to the right thing, but please just use another line to assign option before the check.

Reply:
Who is the boff around here? 🤣
Maybe you want to rewrite all the nasty Lua macros as well?

Reply:
I just mean to say that I think
```
option = lua_tounsigned(L, 2);
luaL_argcheck(L, 2, option <= 4, "wrong option value" );
```
looks better.

Reply:
So do I

Reply:
Done